src/Registry.sol

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+/**
+@title Simple Name Registry
+@author hashedMae
+@notice A simple name registry that stores the data on chain
+ */
+contract Registry {
+
+
+    ///@notice Mapping of a name to address of currentOwner
+     mapping(string => address) public nameToUser;
+
+    ///@dev events for tracking when names are registered or released
+    event NameRegistered(string name, address user);
+    event NameReleased(string name, address user);
+
+    /**
+    @notice checks if a name is currently registered to a user
+    @dev for internal use only, not publicly callable
+    @param _name string that is being checked if it is registered
+    @return true if name is currently registered to a user, otherwise false
+     */
+     function _isRegistered(string memory _name) internal view returns(bool) {
```

Comment:
```suggestion
     function _isRegistered(string memory name) internal view returns(bool) {
```
We don't need to use underscores in all parameters. I've found that it is enough to use them in constructors and when copying to memory a struct.